core/database.py

```
@@ -19,11 +19,23 @@ async def init_db():
+        new_pool = await asyncpg.create_pool(POSTGRES_URL)
+        pool = new_pool
         logger.info("Database connection pool created.")
         await create_schema()
     except Exception as e:
+        if new_pool is not None:
+            try:
+                await new_pool.close()
+            finally:
+                if pool is new_pool:
+                    pool = None
         logger.error(f"Failed to initialize database: {e}")
```

Ruff (0.15.2)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)